### PR TITLE
Removed Unused Imports

### PR DESCRIPTION
**Pull request by aronbos**
Remove all unused import is a code style update and does not break anything submitted to develop branch.

No tests are run.

### CODE REVIEW

**Comment by luke-** [comment]
Thanks for the contribution.

A few requests:

- Can you please commit it in smaller bunches (e.g. per module), so it's easier to review for us.
- Please make sure that all authors have accepted our CLA.
- Currently a test is fails, which indicates a problem with a removed import.

**Comment by ArchBlood** [comment]
Seems that the `MailSummaryProcessor` has a missing call to `MailSummary` from looking it over, so I'm sure there are other class calls that are still being used throughout this PR

File: protected/humhub/modules/admin/controllers/UserPermissionsController.php [hunk 1]
```diff
@@ -11,13 +11,10 @@
 use humhub\modules\admin\permissions\ManageSettings;
 use humhub\modules\admin\permissions\ManageUsers;
 use humhub\modules\content\components\ContentContainerDefaultPermissionManager;
-use humhub\modules\content\models\ContentContainerDefaultPermission;
```
Comment (ArchBlood) [inline protected/humhub/modules/admin/controllers/UserPermissionsController.php:14]: This is still used

File: protected/humhub/modules/admin/grid/SpaceImageColumn.php [hunk 2]
```diff
@@ -9,7 +9,6 @@
 namespace humhub\modules\admin\grid;
 
 use humhub\modules\space\widgets\Image as SpaceImage;
-use humhub\modules\space\models\Space;
```
Comment (ArchBlood) [inline protected/humhub/modules/admin/grid/SpaceImageColumn.php:12]: `$model` = Space class call

File: protected/humhub/modules/admin/grid/SpaceActionColumn.php [hunk 3]
```diff
@@ -10,8 +10,6 @@
 
 use Yii;
 use humhub\libs\ActionColumn;
-use humhub\modules\space\models\Space;
```
Comment (ArchBlood) [inline protected/humhub/modules/admin/grid/SpaceActionColumn.php:13]: `$model` = Space class call

File: protected/humhub/modules/admin/models/forms/UserEditForm.php [hunk 4]
```diff
@@ -2,6 +2,10 @@
 
 namespace humhub\modules\admin\models\forms;
 
+use humhub\modules\user\models\GroupUser;
+use Yii;
+use humhub\modules\user\models\User;
+use humhub\modules\user\models\Group;
```
Comment (ArchBlood) [inline protected/humhub/modules/admin/models/forms/UserEditForm.php:8]: Not sure if it's on GitHub mobile app but shows two `Group` class calls

File: protected/humhub/modules/admin/views/setting/statistic.php [hunk 5]
```diff
@@ -4,7 +4,6 @@
 use yii\helpers\Html;
 use humhub\modules\admin\models\forms\StatisticSettingsForm;
 use humhub\modules\ui\form\widgets\CodeMirrorInputWidget;
-use yii\web\View;
```
Comment (ArchBlood) [inline protected/humhub/modules/admin/views/setting/statistic.php:7]: Should use the new UI view here instead.

File: protected/humhub/modules/content/notifications/ContentDeleted.php [hunk 6]
```diff
@@ -11,9 +11,6 @@
 use humhub\modules\notification\components\BaseNotification;
 use Yii;
 use yii\bootstrap\Html;
-use humhub\modules\user\models\User;
-use humhub\libs\Helpers;
-use yii\helpers\Json;
```
Comment (ArchBlood) [inline protected/humhub/modules/content/notifications/ContentDeleted.php:16]: Again `$this->payload` = JSON?

File: protected/humhub/modules/content/notifications/ContentDeleted.php [hunk 7]
```diff
@@ -11,9 +11,6 @@
 use humhub\modules\notification\components\BaseNotification;
 use Yii;
 use yii\bootstrap\Html;
-use humhub\modules\user\models\User;
```
Comment (ArchBlood) [inline protected/humhub/modules/content/notifications/ContentDeleted.php:14]: `$this->originator->displayName` = User?

**Review by ArchBlood — COMMENTED** [review]
Just a few things I've looked at, but seeing how much is changed I'd recommend opening an issue then breaking this PR down into smaller PRs.

File: protected/humhub/modules/admin/controllers/UserPermissionsController.php (hunk 1 above)
Comment (aronbos) [reply protected/humhub/modules/admin/controllers/UserPermissionsController.php:14]: To be investigated. Can not find it.

File: protected/humhub/libs/ActionColumn.php [hunk 8]
```diff
@@ -11,7 +11,6 @@
 use humhub\widgets\Button;
 use yii\grid\Column;
 use yii\helpers\Url;
-use humhub\libs\Html;
```
[the remark this answers is not in the document]
Comment (aronbos) [reply protected/humhub/libs/ActionColumn.php:14]: You are correct. It should not be removed.

File: protected/humhub/modules/activity/components/MailSummaryProcessor.php [hunk 9]
```diff
@@ -11,7 +11,6 @@
 use humhub\modules\activity\Module;
 use Yii;
 use yii\helpers\Console;
-use humhub\modules\activity\components\MailSummary;
```
[the remark this answers is not in the document]
Comment (aronbos) [reply protected/humhub/modules/activity/components/MailSummaryProcessor.php:14]: You are correct. It should not be removed.

File: protected/humhub/modules/admin/controllers/UserPermissionsController.php [hunk 10]
```diff
@@ -11,13 +11,10 @@
...
 use humhub\modules\content\models\ContentContainerPermission;
-use humhub\modules\content\models\ContentContainerSetting;
```
[the remark this answers is not in the document]
Comment (aronbos) [reply protected/humhub/modules/admin/controllers/UserPermissionsController.php:16]: To be investigated. Can not find it.

File: protected/humhub/modules/admin/grid/SpaceActionColumn.php (hunk 3 above)
Comment (aronbos) [reply protected/humhub/modules/admin/grid/SpaceActionColumn.php:13]: To be investigated. Can not find it. Are we using different versions of Humhub?

Comment (aronbos) [reply protected/humhub/modules/admin/grid/SpaceActionColumn.php:13]: Is this the best way to solve this problem?  How do I know the relation between Space and $model?

File: protected/humhub/modules/admin/grid/SpaceImageColumn.php (hunk 2 above)
Comment (aronbos) [reply protected/humhub/modules/admin/grid/SpaceImageColumn.php:12]: Is this the best way to solve this problem? How do I know the relation between Space and $model?

File: protected/humhub/modules/comment/notifications/CommentDeleted.php [hunk 11]
```diff
@@ -9,15 +9,9 @@
 namespace humhub\modules\comment\notifications;
 
 use humhub\modules\comment\models\Comment;
-use humhub\modules\content\components\ContentActiveRecord;
-use humhub\modules\content\widgets\richtext\converter\RichTextToShortTextConverter;
 use humhub\modules\notification\components\BaseNotification;
-use humhub\modules\notification\models\Notification;
-use humhub\modules\user\models\User;
```
[the remark this answers is not in the document]
Comment (aronbos) [reply protected/humhub/modules/comment/notifications/CommentDeleted.php:16]: Your ? is the problem in the code. How to we know the relation?

File: protected/humhub/modules/comment/notifications/CommentDeleted.php [hunk 12]
```diff
@@ -9,15 +9,9 @@
...
-use humhub\modules\user\notifications\Mentioned;
 use Yii;
 use yii\bootstrap\Html;
-use yii\helpers\Json;
```
[the remark this answers is not in the document]
Comment (aronbos) [reply protected/humhub/modules/comment/notifications/CommentDeleted.php:20]: Your ? is the problem in the code. How to we know the relation?

File: protected/humhub/modules/content/notifications/ContentDeleted.php (hunk 7 above)
Comment (aronbos) [reply protected/humhub/modules/content/notifications/ContentDeleted.php:14]: Your ? is the problem in the code. How to we know the relation?

File: protected/humhub/modules/content/notifications/ContentDeleted.php (hunk 6 above)
Comment (aronbos) [reply protected/humhub/modules/content/notifications/ContentDeleted.php:16]: Your ? is the problem in the code. How to we know the relation?

Comment (ArchBlood) [reply protected/humhub/modules/content/notifications/ContentDeleted.php:16]: `$this->payload` refers to a JSON string

File: protected/humhub/modules/comment/notifications/CommentDeleted.php (hunk 11 above)
[the remark this answers is not in the document]
Comment (ArchBlood) [reply protected/humhub/modules/comment/notifications/CommentDeleted.php:16]: See https://github.com/humhub/humhub/pull/6245#discussion_r1171785274

File: protected/humhub/modules/comment/notifications/CommentDeleted.php (hunk 12 above)
[the remark this answers is not in the document]
Comment (ArchBlood) [reply protected/humhub/modules/comment/notifications/CommentDeleted.php:20]: See https://github.com/humhub/humhub/pull/6245#discussion_r1171786259

File: protected/humhub/modules/admin/grid/SpaceImageColumn.php (hunk 2 above)
Comment (ArchBlood) [reply protected/humhub/modules/admin/grid/SpaceImageColumn.php:12]: `getSpace()` is called
https://github.com/humhub/humhub/blob/d79fc2010c18cc2ccda0f05e0f28235dece058d4/protected/humhub/modules/admin/grid/SpaceImageColumn.php#L34-L37

File: protected/humhub/modules/admin/models/forms/UserEditForm.php (hunk 4 above)
Comment (ArchBlood) [reply protected/humhub/modules/admin/models/forms/UserEditForm.php:8]: Yes, this does in fact add the `Group` class twice.

File: protected/humhub/modules/admin/grid/SpaceImageColumn.php (hunk 2 above)
Comment (aronbos) [reply protected/humhub/modules/admin/grid/SpaceImageColumn.php:12]: I am no expert in PHP. So please correct me if I am wrong.
$this refers to the class SpaceImageColumn. Or to ...? 
$model is variable with no reference to the use. How should PHP make a link between both?
So why do we need use humhub\modules\space\models\Space;?

**Comment by aronbos** [comment]
There is a difference between an use and a cast. We can not use a use for a cast. The minimum version of Humhub is 7.4, so we can cast objects. Adding casts to the software would clear up our discussion.

**Pull request closed without merging**